src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -374,7 +398,7 @@ public enum ChunkVersions : byte {
 			var md5 = MD5.Create();
 
 			// ALLOCATE MEM
-			Interlocked.Exchange(ref _isCached, 1);
+			_cacheStatus = CacheStatus.Cached;
 			_cachedLength = fileSize;
 			_cachedData = Marshal.AllocHGlobal(_cachedLength);
```

Comment:
do we also need to lock the CreateInMemChunk contents with a _cachedDataLock?

Reply:
i dont _think_ so, because inmem doesn't do any uncaching/recaching, it just caches when creating the chunk and keeps it cached

Reply:
fair enough :+1: 

i was just thinking if it would make sense if we move the `_cacheStatus = CacheStatus.Cached` assignment further below in the call after we have created the memstreams just to be consistent with acquire-release semantics. i.e if later on we make a change where another thread reads _cacheStatus = Cached (say outside the _cachedDataLock for some reason), then we can be sure that it has been cached.